common/djangoapps/enrollment/views.py

```diff
@@ -616,13 +616,14 @@ def post(self, request):
                 org = course_id.org
                 update_email_opt_in(request.user, org, email_opt_in)
 
+            log.info('The user %s has already been enrolled in course run %s.', username, course_id)
             return Response(response)
         except CourseModeNotFoundError as error:
             return Response(
                 status=status.HTTP_400_BAD_REQUEST,
                 data={
                     "message": (
-                        u"The course mode '{mode}' is not available for course '{course_id}'."
+                        u"The [{mode}] course mode is expired or otherwise unavailable for course run [{course_id}]."
```

Comment:
@schenedx this message will make someone not intimately familiar with our system think that the course mode was missing for the course. It completely precludes the (more common) possibility that the course mode expired, and makes debugging more difficult. This message is an improvement.

Again, see our [previous attempt](https://github.com/edx/edx-platform/pull/13663/files#diff-a4a6f81c03f035ebb4441f342e27c62eR633) to fix this.